### PR TITLE
Rss ecomm 4 07 08 detailed product page implementation

### PR DESCRIPTION
**Pull request by Pavel-Kuvshinov**
## Overview
Added detailed product page implementation.

**What has been done:**
- Fix styles in header navigation;
- Add Product Duplication check; 
- Add remove from cart  and add to cart abilities for the detailed product page

**Why this has been done:**
- These changes added detailed product page implementation.

**Issue(s) closed:**
- [Issue RSS-ECOMM-4_07: Add Product to Cart Functionality for Non-Duplicated Products](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint4/RSS-ECOMM-4_07.md);
- [Issue RSS-ECOMM-4_08: Remove Product from Cart Functionality](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint4/RSS-ECOMM-4_08.md).

## Type of Change
- [x] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: kickFlip-project/src/pages/product/productPage.tsx [hunk 1]
```diff
@@ -19,6 +23,28 @@ export default function ProductPage() {
     const [productData, setProductData] = useState<ProductResponse | null>(null);
     const [productError, setProductErrorError] = useState('');
 
+    const [cartData, setCartData] = useState<CartResponse | null>();
```
Comment (makarovaiuliia) [inline kickFlip-project/src/pages/product/productPage.tsx]: Зачем мы это делаем, если у нас есть данные корзины в сторе? и мы их можем достать через const itemsInCart = useSelector(getCartItems);

File: kickFlip-project/src/pages/product/productPage.tsx [hunk 2]
```diff
@@ -19,6 +23,28 @@ export default function ProductPage() {
...
+    const [cartError, setCartError] = useState('');
+
+    const idCart: string = useSelector(getCartId);
+
+    useEffect(() => {
+        const fetchCartItems = async () => {
```
Comment (makarovaiuliia) [inline kickFlip-project/src/pages/product/productPage.tsx]: Запрос не нужен, если тебе нужны только данные о том, что есть в корзине

File: kickFlip-project/src/components/product/productDetails/productDetailsContainer.tsx [hunk 3]
```diff
@@ -1,18 +1,67 @@
...
+                {
+                    action: updateAction,
+                    lineItemId: cartData.lineItems[cartItemIndex].id,
+                    quantity,
+                },
+            ],
+        };
+        try {
+            const newCart = await updateCart(`${cartId}`, changedData);
+            setCartData(newCart);
```
Comment (makarovaiuliia) [inline kickFlip-project/src/components/product/productDetails/productDetailsContainer.tsx]: вот от этого тоже можно будет избавится

File: kickFlip-project/src/components/product/productDetails/productDetailsContainer.tsx [hunk 4]
```diff
@@ -1,18 +1,67 @@
...
+            dispatch(setCart(newCart));
```
Comment (makarovaiuliia) [inline kickFlip-project/src/components/product/productDetails/productDetailsContainer.tsx]: Вот это не понимаю зачем в компоненте - если после updateCart нужно поставить новые данные корзины, то делать это нужно в слайсе в ExtraReducers

**Review by makarovaiuliia — CHANGES_REQUESTED** [review]
Небольшие комменты по коду

File: kickFlip-project/src/pages/product/productPage.tsx (hunk 1 above)
Comment (Pavel-Kuvshinov) [reply kickFlip-project/src/pages/product/productPage.tsx]: Исправил

File: kickFlip-project/src/pages/product/productPage.tsx (hunk 2 above)
Comment (Pavel-Kuvshinov) [reply kickFlip-project/src/pages/product/productPage.tsx]: Исправил

File: kickFlip-project/src/components/product/productDetails/productDetailsContainer.tsx (hunk 3 above)
Comment (Pavel-Kuvshinov) [reply kickFlip-project/src/components/product/productDetails/productDetailsContainer.tsx]: Исправил

File: kickFlip-project/src/components/product/productDetails/productDetailsContainer.tsx (hunk 4 above)
Comment (Pavel-Kuvshinov) [reply kickFlip-project/src/components/product/productDetails/productDetailsContainer.tsx]: Исправил

**Review by makarovaiuliia — APPROVED** [review]

**Pull request merged**